site/docs/plugins/i18n.md

```diff
@@ -1,105 +1,751 @@
+For instance, take a look at the above example using sessions.
+
+When you only do
+
+```ts:no-line-numbers
+ctx.session.__language_code = "de";
+```
+
+It will not update the currently used locale in the i18n instance.
+Instead, it only updates the session.
+Thus, the changes will only take place _for the next update_.
+
+If you cannot wait until the next update, you might need to refresh the changes after updating the user language.
+Use the `renegotiateLocale` method for these cases.
+
+```ts
+ctx.session.__language_code = "de";
+await ctx.i18n.renegotiateLocale();
+```
+
+Anyplace we use the method `t`, the bot will try to reply with the German translation of that message (specified in `locales/de.ftl`).
+
+Also, remember that when you use built-in sessions, you can achieve the same result using the `setLocale` method.
+
+:::
+
+::: tip Setting the Locale When Not Using Sessions
+
+When [not using sessions](#without-sessions), if there is a case where you need to set the locale for a user, you can do that by using the `useLocale` method.
+
+```ts
+await ctx.i18n.useLocale("de");
+```
+
+It sets the specified locale to be used for future translations.
+The effect lasts only for the current update and is not preserved.
+You can use this method to change the translation locale in the middle of the update (e.g., when the user changes the language).
+
+:::
+
+## Custom Locale Negotiation
+
+You can use the option `localeNegotiator` to specify a custom locale negotiator.
+This option is helpful if you want to select the locale based on external sources (such as databases) or in other situations where you want to control which locale is used.
+
+Here is the default order of how the plugin chooses its locale:
+
+1. If sessions are enabled, try to read `__language_code` from the session.
+   If it returns a valid locale, it is used.
+   If it returns nothing or a non-registered locale, move on to step 2.
+2. Try to read from `ctx.from.language_code`.
+   If it returns a valid locale, it is used.
+   If it returns nothing or a non-registered locale, move on to step 3.
+
+   > Note that `ctx.from.language_code` is only available if the user has started the bot.
+   > That means if the bot sees the user in a group or somewhere without the user previously having started the bot, it won't be able to see `ctx.from.language_code`.
+3. Try using the default language configured in the options of `I18n`.
+   If it is set to a valid locale, it is used.
+   If it isn't specified or set to a non-registered locale, move on to step 4.
+4. English (`en`). The plugin itself sets this as the ultimate fallback locale.
+   Even though it is a fallback locale, and we recommend having a translation, it is not a requirement.
+   If no English locale is provided, move on to step 5.
+5. If all the above things fail, use `{key}` instead of a translation. We **highly recommend** setting a locale that exists in your translations as `defaultLocale` in i18n options.
```

Comment:
```suggestion
5. If all the above things fail, use `{key}` instead of a translation.
   We **highly recommend** setting a locale that exists in your translations as `defaultLocale` in the `I18n` options.
```